File: Array/2545.Sort_the_Students_by_Their_Kth_Score/C++/2545.Sort_the_Students_by_Their_Kth_Score.cpp
class Solution {
public:
    vector<vector<int>> sortTheStudents(vector<vector<int>> & score, int k) 
    {
        //=============================================//
        // Input type:                                 //
        //   - vector<vector<int>> ref. (score vector) //
        //   - int (k-th index)                        //
        // Return type:                                //
        //   - vector<vector<int>> (result vector)     //
        //=============================================//

        /*Initialize*/
        ///// Length of rows /////
        int len_rows = score.size();

        ///// Record vector /////
        vector<pair<int, int>> record_vect;

        ///// Result vector /////
        vector<vector<int>> res_vect;


        /*Sorted-based loop traversal with recorded vector*/
        ///// Step 1: Record indexed-value informations with vector /////
        for (int rows_idx = 0; (rows_idx < len_rows); (rows_idx++)) //Rows
        {
            record_vect.emplace_back((score[rows_idx])[k], rows_idx); //Keep updating, recording

        } //Rows

        ///// Step 2: Looped-traversal for the recorded vector /////
        sort(record_vect.begin(), record_vect.end(), [&](const pair<int, int> & pair_1, const pair<int, int> & pair_2)
        {
            return (pair_1.first > pair_2.first);

        }); //Sorted-operations

        for (int rows_idx = 0; (rows_idx < len_rows); (rows_idx++)) //Rows
        {
            res_vect.push_back(score[(record_vect[rows_idx]).second]); //Keep updating, recording

        } //Rows

        return res_vect;
    }
};

File: Array/2545.Sort_the_Students_by_Their_Kth_Score/Python/2545.Sort_the_Students_by_Their_Kth_Score.py
class Solution(object):
    def sortTheStudents(self, score, k):
        """
        :type score: List[List[int]]
        :type k: int
        :rtype: List[List[int]]
        """
        #===============================#
        # Sorted-based traversal method #
        #===============================#

        ############
        #Initialize
        ##### Length of rows #####
        len_rows = len(score)

        ##### Record array #####
        record_arry = []

        ##### Result array #####
        res_arry = []


        #################################################
        #Sorted-based loop traversal with recorded array
        ##### Step 1: Record indexed-value informations with array #####
        for rows_idx in range(len_rows):
            record_arry.append([(score[rows_idx])[k], rows_idx]) #Keep updating/recording

        ##### Step 2: Looped-traversal for the recorded array #####
        record_arry.sort(reverse = True) #Sorted-operations

        for rows_idx in range(len_rows):
            res_arry.append(score[(record_arry[rows_idx])[1]]) #Keep updating/recording

        return res_arry
